File: src/Practice/CozyCrawler.Runner/BlockedUrl2ResultRunner.cs
using CozyCrawler.Interface;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace CozyCrawler.Runner
{
    public class BlockedUrl2ResultRunner : IUrl2ResultRunner
    {
        IUrl2Result trans_;

        public void To(IUrl2Result to)
        {
            trans_ = to;
        }

        public void OnNewUrl(string url)
        {
            trans_?.OnNewUrl(url);
        }

        public void Start()
        {
        }

        public void Stop()
        {
        }
    }
}

File: src/Practice/CozySeeker/Runner/BlockedUrl2ResultRunner.cpp
#include "BlockedUrl2ResultRunner.h"

NS_BEGIN

BlockedUrl2ResultRunner::BlockedUrl2ResultRunner()
{

}

BlockedUrl2ResultRunner::~BlockedUrl2ResultRunner()
{

}

void BlockedUrl2ResultRunner::OnNewUrl(StrPtr url)
{
    if (m_trans != nullptr)
    {
        m_trans->OnNewUrl(url);
    }
}

void BlockedUrl2ResultRunner::To(IUrl2ResultPtr to)
{
    m_trans = to;
}

void BlockedUrl2ResultRunner::Start()
{

}

void BlockedUrl2ResultRunner::Stop()
{

}

NS_END